### PR TITLE
Display warning message if no known progress logging frontend is loaded

### PR DESCRIPTION
**Pull request by devmotion**
This PR is a take on https://github.com/TuringLang/AbstractMCMC.jl/issues/18#issuecomment-594759943. I'm not sure yet how reliable this approach detects if one of the known and recommend progress logging frontends is set up, or not.

### CODE REVIEW

**Comment by devmotion** [comment]
Not completely sure how to test this though, maybe one has to use SafeTestsets or manually create some modules?

File: src/AbstractMCMC.jl [hunk 1]
```diff
@@ -77,6 +92,9 @@ function StatsBase.sample(
     # Perform any necessary setup.
     sample_init!(rng, model, sampler, N; kwargs...)
 
+    # Check progress logger frontends.
+    progress && checkprogresslogger()
```
Comment (cpfiffer) [inline src/AbstractMCMC.jl]: This doesn't seem like `checkprogresslogger()` will be run unless `progress==false`. Should this be `!progress && checkprogresslogger()`?

**Comment by cpfiffer** [comment]
> Not completely sure how to test this though, maybe one has to use SafeTestsets or manually create some modules?

Maybe

```julia
@safetestset "Warning test" begin 
    @test_logs (:warn, warning_message) using AbstractMCMC 
end
```

Although I'm not sure if this will scope correctly.

File: src/AbstractMCMC.jl (hunk 1 above)
Comment (devmotion) [reply src/AbstractMCMC.jl]: It performs the check if `progress` logging is enabled - that seems reasonable, doesn't it? If progress logging is disabled we do not care about the loaded packages, I guess.

Comment (cpfiffer) [reply src/AbstractMCMC.jl]: Woops. Misread `&` as `|` somehow.

**Pull request closed without merging**